### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by domanttas**
Implemented game logic for getting and shooting targets, design is just for testing purposes, it looks bad and concept of "good frontend code" was not applied because I quite dislike frontend...

Created few API calls;
Created custom class with Message property as .NET bad request returns object, not a string as in .NET Core;
Initial logic of whole game flow;
Initial display of game.

### CODE REVIEW

File: WhoIsThat/WhoIsThat/Connections/IRestService.cs [hunk 1]
```diff
@@ -38,5 +38,34 @@ public interface IRestService
         /// <param name="user">User to insert</param>
         /// <returns>True if user was inserted</returns>
         Task<bool> InsertUserIntoRecognition(ImageObject user);
+
+        /// <summary>
+        /// Assigns random target
+        /// </summary>
+        /// <param name="id">ID of current user</param>
+        /// <returns>Assigned target's ID</returns>
+        Task<int> GetRandomTarget(int id);
```
Comment (tomasmuzas) [inline WhoIsThat/WhoIsThat/Connections/IRestService.cs]: Not sure I get this?

Comment (tomasmuzas) [reply WhoIsThat/WhoIsThat/Connections/IRestService.cs]: Oh, okay, understood:D but id is kind of cryptic

Comment (tomasmuzas) [reply WhoIsThat/WhoIsThat/Connections/IRestService.cs]: Better use more descriptive names like `userId`, `hunterId` etc

File: WhoIsThat/WhoIsThat/Connections/RestService.cs [hunk 2]
```diff
@@ -33,7 +34,8 @@ public async Task<List<ImageObject>> GetImageObjects()
             
             if (!response.IsSuccessStatusCode)
             {
-                throw new ManagerException(await response.Content.ReadAsStringAsync());
+                var errorContent = await response.Content.ReadAsStringAsync();
+                throw new ManagerException((JsonConvert.DeserializeObject<BadRequestModel>(errorContent)).Message);
```
Comment (tomasmuzas) [inline WhoIsThat/WhoIsThat/Connections/RestService.cs]: Thats nice!! Just split it into multiple lines:)

File: WhoIsThat/WhoIsThat/ViewModels/HomeViewModel.cs [hunk 3]
```diff
@@ -22,85 +24,186 @@ public class HomeViewModel : INotifyPropertyChanged
...
+                    OnPropertyChanged("DisplayStatus");
+
+                    User = await _restService.UpdateUserScore(User.Id);
+
+                    Name = "Welcome, " + User.PersonFirstName + ". Score: " + User.Score.ToString();
+                    OnPropertyChanged("Name");
+                }
+            }
+
+            catch (ManagerException noFacesFoundException) when (noFacesFoundException.ErrorCode == Constants.NoFacesIdentifiedError)
```
Comment (tomasmuzas) [inline WhoIsThat/WhoIsThat/ViewModels/HomeViewModel.cs]: You could probably shorten it up to `ex` :D

Comment (domanttas) [reply WhoIsThat/WhoIsThat/ViewModels/HomeViewModel.cs]: I tried to make it look serious, but yeah, I could definitely do that :D

File: WhoIsThat/WhoIsThat/Connections/RestService.cs (hunk 2 above)
Comment (domanttas) [reply WhoIsThat/WhoIsThat/Connections/RestService.cs]: Will do that, next branch will be called "Cleaning" or something :D

**Comment by domanttas** [comment]
@tomasmuzas Thanks for the review, appreciated as always! :)

File: WhoIsThat/WhoIsThat/Connections/RestService.cs (hunk 2 above)
Comment (tomasmuzas) [reply WhoIsThat/WhoIsThat/Connections/RestService.cs]: Refactoring is the word;)

Comment (domanttas) [reply WhoIsThat/WhoIsThat/Connections/RestService.cs]: Amen :)

**Pull request merged**